docs/troubleshooting/monitoring.md

```diff
@@ -316,3 +316,277 @@ You may encounter this when you install the Harvester v1.3.0 or higher version c
+subjects:
+- kind: ServiceAccount
+  name: rancher-monitoring-crd-manager
+  namespace: cattle-monitoring-system
+
+$ kubectl get configmap -n cattle-monitoring-system rancher-monitoring-crd-manifest
+apiVersion: v1
+data:
+  crd-manifest.tgz.b64: ...
+kind: ConfigMap
+metadata:
+  annotations:
+    meta.helm.sh/release-name: rancher-monitoring-crd
+    meta.helm.sh/release-namespace: cattle-monitoring-system
+  creationTimestamp: "2023-01-09T11:04:33Z"
+  labels:
+    app.kubernetes.io/managed-by: Helm
+  name: rancher-monitoring-crd-manifest
+  namespace: cattle-monitoring-system
+  ...
+
+$ kubectl get ServiceAccount -n cattle-monitoring-system rancher-monitoring-crd-manager
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    meta.helm.sh/release-name: rancher-monitoring-crd
+    meta.helm.sh/release-namespace: cattle-monitoring-system
+  creationTimestamp: "2023-01-09T11:04:33Z"
+  labels:
+    app: rancher-monitoring-crd-manager
+    app.kubernetes.io/managed-by: Helm
+  name: rancher-monitoring-crd-manager
+  namespace: cattle-monitoring-system
+  ...
+```
+
+### Root Cause
+
+The objects that are marked as `missing` do not have the related annotations and labels required by the `ManagedChart` object.
+
+Example:
+
+```
+One of the manually recreated object:
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  annotations:
+    meta.helm.sh/release-name: rancher-monitoring-crd
+    meta.helm.sh/release-namespace: cattle-monitoring-system
+    objectset.rio.cattle.io/id: default-mcc-rancher-monitoring-crd-cattle-fleet-local-system   # This required item is not in the above object.
+  creationTimestamp: "2024-04-03T10:23:55Z"
+  labels:
+    app: rancher-monitoring-crd-manager
+    app.kubernetes.io/managed-by: Helm
+    objectset.rio.cattle.io/hash: 2da503261617e9ea2da822d2da7cdcfccad847a9    # This required item is not in the above object.
+  name: rancher-monitoring-crd-manager
+...
+rules:
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - get
+  - patch
+  - delete
+  - update
+```
+
+### Workaround
+
+1. Patch ClusterRole object `rancher-monitoring-crd-manager` to add the `update` operation.
+
+    ```
+    $ cat > patchrules.yaml << EOF
+    rules:
+    - apiGroups:
+      - apiextensions.k8s.io
+      resources:
+      - customresourcedefinitions
+      verbs:
+      - create
+      - get
+      - patch
+      - delete
+      - update
+    EOF
+
+    $ kubectl patch ClusterRole rancher-monitoring-crd-manager --patch-file ./patchrules.yaml --type merge
+
+    $ rm ./patchrules.yaml
+    ```
+
+1. Patch the missing objects with the required annotations and labels.
+
+    ```
+    $ cat > patchhash.yaml << EOF
+    metadata:
+      annotations:
+        objectset.rio.cattle.io/id: default-mcc-rancher-monitoring-crd-cattle-fleet-local-system
+      labels:
+        objectset.rio.cattle.io/hash: 2da503261617e9ea2da822d2da7cdcfccad847a9
```

Comment:
If we could explain how this hash value is calculated, users who are applying the patch would be more confident. Is this value always the same for every cluster?

Reply:
Explained  https://github.com/harvester/docs/pull/541#issue-2223364760

Reply:
Oh cool, didn't noticed that. Thanks for the explanation.